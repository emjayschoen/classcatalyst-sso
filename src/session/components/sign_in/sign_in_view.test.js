import React from 'react';
import { shallow } from 'enzyme/build';
import SignInView from './sign_in_view';

const FOO_LOCATION = { state: { returnTo: '/foo/bar' } };
const NO_LOCATION = { state: {} };

describe('<SignInView />', () => {
  let subject;

  describe('not logged in', () => {
    describe('without error', () => {
      beforeEach(() => {
        subject = shallow(<SignInView signedIn={false} handleSubmit={jest.fn()} location={FOO_LOCATION} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('without location set', () => {
      beforeEach(() => {
        subject = shallow(<SignInView signedIn={false} handleSubmit={jest.fn()} location={NO_LOCATION} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('with error', () => {
      beforeEach(() => {
        subject = shallow(<SignInView signedIn={false} handleSubmit={jest.fn()} location={FOO_LOCATION} status="failed" />);
      });
      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });

  describe('when already logged in', () => {
    describe('with a returnTo location state set', () => {
      beforeEach(() => {
        subject = shallow(<SignInView signedIn handleSubmit={jest.fn()} location={FOO_LOCATION} status={undefined} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
    describe('without a returnTo location state set', () => {
      beforeEach(() => {
        subject = shallow(<SignInView signedIn handleSubmit={jest.fn()} location={NO_LOCATION} status={undefined} />);
      });

      it('renders correctly', () => {
        expect(subject).toMatchSnapshot();
      });
    });
  });
});
