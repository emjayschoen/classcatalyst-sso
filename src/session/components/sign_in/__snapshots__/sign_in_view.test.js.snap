// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInView /> not logged in with error renders correctly 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Sign In
    </h3>
    <Field
      component={[Function]}
      fullWidth={true}
      label="Email"
      name="email"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Password"
      name="password"
      required={true}
      type="password"
    />
    <input
      className="btn"
      disabled={false}
      type="submit"
      value="Login"
    />
    <span
      className="status"
    />
    <div
      className="oauth-row"
    >
      <span>
        Sign in with:
      </span>
      <a
        className="oauth-button"
        href="/mockauth/linkedin/%2Ffoo%2Fbar"
        key="linkedin"
      >
        <i
          className="fab fa-linkedin fa-5x"
        />
      </a>
      <a
        className="oauth-button"
        href="/mockauth/facebook/%2Ffoo%2Fbar"
        key="facebook"
      >
        <i
          className="fab fa-facebook fa-5x"
        />
      </a>
      <a
        className="oauth-button"
        href="/mockauth/google/%2Ffoo%2Fbar"
        key="google"
      >
        <i
          className="fab fa-google fa-5x"
        />
      </a>
      <a
        className="oauth-button"
        href="/mockauth/twitter/%2Ffoo%2Fbar"
        key="twitter"
      >
        <i
          className="fab fa-twitter fa-5x"
        />
      </a>
    </div>
    <p>
      <Link
        to="/forgot"
      >
        Lost your password?
      </Link>
      <span
        style={
          Object {
            "padding": "0 1em",
          }
        }
      >
        |
      </span>
      <Link
        to="/register"
      >
        Don't have an account?
      </Link>
    </p>
  </form>
</styled.div>
`;

exports[`<SignInView /> not logged in without error renders correctly 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Sign In
    </h3>
    <Field
      component={[Function]}
      fullWidth={true}
      label="Email"
      name="email"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Password"
      name="password"
      required={true}
      type="password"
    />
    <input
      className="btn"
      disabled={false}
      type="submit"
      value="Login"
    />
    <span
      className="status"
    />
    <div
      className="oauth-row"
    >
      <span>
        Sign in with:
      </span>
      <a
        className="oauth-button"
        href="/mockauth/linkedin/%2Ffoo%2Fbar"
        key="linkedin"
      >
        <i
          className="fab fa-linkedin fa-5x"
        />
      </a>
      <a
        className="oauth-button"
        href="/mockauth/facebook/%2Ffoo%2Fbar"
        key="facebook"
      >
        <i
          className="fab fa-facebook fa-5x"
        />
      </a>
      <a
        className="oauth-button"
        href="/mockauth/google/%2Ffoo%2Fbar"
        key="google"
      >
        <i
          className="fab fa-google fa-5x"
        />
      </a>
      <a
        className="oauth-button"
        href="/mockauth/twitter/%2Ffoo%2Fbar"
        key="twitter"
      >
        <i
          className="fab fa-twitter fa-5x"
        />
      </a>
    </div>
    <p>
      <Link
        to="/forgot"
      >
        Lost your password?
      </Link>
      <span
        style={
          Object {
            "padding": "0 1em",
          }
        }
      >
        |
      </span>
      <Link
        to="/register"
      >
        Don't have an account?
      </Link>
    </p>
  </form>
</styled.div>
`;

exports[`<SignInView /> not logged in without location set renders correctly 1`] = `
<styled.div>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Sign In
    </h3>
    <Field
      component={[Function]}
      fullWidth={true}
      label="Email"
      name="email"
      required={true}
      type="text"
    />
    <Field
      component={[Function]}
      fullWidth={true}
      label="Password"
      name="password"
      required={true}
      type="password"
    />
    <input
      className="btn"
      disabled={false}
      type="submit"
      value="Login"
    />
    <span
      className="status"
    />
    <div
      className="oauth-row"
    >
      <span>
        Sign in with:
      </span>
      <a
        className="oauth-button"
        href="/mockauth/linkedin/%2Fdashboard"
        key="linkedin"
      >
        <i
          className="fab fa-linkedin fa-5x"
        />
      </a>
      <a
        className="oauth-button"
        href="/mockauth/facebook/%2Fdashboard"
        key="facebook"
      >
        <i
          className="fab fa-facebook fa-5x"
        />
      </a>
      <a
        className="oauth-button"
        href="/mockauth/google/%2Fdashboard"
        key="google"
      >
        <i
          className="fab fa-google fa-5x"
        />
      </a>
      <a
        className="oauth-button"
        href="/mockauth/twitter/%2Fdashboard"
        key="twitter"
      >
        <i
          className="fab fa-twitter fa-5x"
        />
      </a>
    </div>
    <p>
      <Link
        to="/forgot"
      >
        Lost your password?
      </Link>
      <span
        style={
          Object {
            "padding": "0 1em",
          }
        }
      >
        |
      </span>
      <Link
        to="/register"
      >
        Don't have an account?
      </Link>
    </p>
  </form>
</styled.div>
`;

exports[`<SignInView /> when already logged in with a returnTo location state set renders correctly 1`] = `
<Redirect
  to="/foo/bar"
/>
`;

exports[`<SignInView /> when already logged in without a returnTo location state set renders correctly 1`] = `
<Redirect
  to="/dashboard"
/>
`;
